core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -277,12 +278,26 @@ trait SubmissionMonitor extends FutureSupport with LazyLogging with RawlsInstrum
   private def execServiceOutputs(workflowRec: WorkflowRecord, petUser: UserInfo)(implicit executionContext: ExecutionContext): Future[Option[(WorkflowRecord, Option[ExecutionServiceOutputs])]] = {
     WorkflowStatuses.withName(workflowRec.status) match {
       case WorkflowStatuses.Succeeded =>
+        // asynchronously upload metadata to GCS.
+        // Currently, this is only used for hamm, and may potentially be removed from RAWLS if design changes
+        workflowRec.externalId.traverse{
+          workflowId =>
+            uploadMetadataToGCS(workflowRec.submissionId.toString, workflowId, petUser)
```

Comment:
this only runs for Succeeded workflows... I think money spent on failed workflows will also be interesting

Reply:
also since these Futures are not monitored by anything, we should handle and log any errors